pyvista/core/pointset.py

```diff
@@ -919,15 +919,26 @@ def n_lines(self) -> int:
     def n_verts(self) -> int:
         """Return the number of vertices.
 
+        A vertex is a single element cell referencing a single point.
```

Comment:
What I mean by this, is that it is possible that a polyvertex is also included, which is a single cell that references a set of points (in a way that I don't fully understand).

Reply:
This interface on vtkPolyData has always felt odd to me... I'm not sure I have a great answer for your first question. Indeed we're just following the VTK data model here.

> is that it is possible that a polyvertex is also included, which is a single cell that references a set of points (in a way that I don't fully understand).

I'm not sure I follow what you're asking? But a vertex cell can only ever point to one Point ID, not a set

Reply:
A vtkVertex is one cell that has one point.  A vtkPolyVertex is one cell that has more than one point, but no topological structure.  That is both are 0D cells.  I'm not sure if `n_verts` here is the sum of all 0D cells or just the vtkVertex ones.  The API isn't explained well enough in vtk for me to know, but most importantly **the description here would not be correct if it includes all 0D cells**.  I can do some testing soon.

This is similar to the difference of vtkLine and vtkPolyLine.

Reply:
Ok, I tested this.  `n_verts` includes _all_ 0D cells, so it is not true that it is just cells with one point.  Our API is especially awkward because we do not allow to instantiate vertices as a separate input.  We need to overhaul this API a bit.

```py
import pyvista as pv
p = [[0, 0, 0], [1, 0, 0]]
m = pv.PolyData(p)  # no way to control 0D cells
m.n_verts # 2
m.verts = [2, 0, 1]  # make 1 vtkPolyVertex
m.n_verts # 1
m.get_cell(0)  # shows it is POLY_VERTEX
```